compiler/src/iree/compiler/Pipelines/Options.h

```diff
@@ -167,9 +167,20 @@ struct SchedulingOptions {
 };
 
 struct PreprocessingOptions {
+  /// Options for converting Linalg matmul ops to transposed variants.
+  enum class TransposeMatmulInput {
+    /// Transpose LHS input matrix.
+    Lhs,
+    /// Transpose RHS input matrix.
+    Rhs,
+    /// Transpose neither input (disable).
+    None
+  };
+
   std::string preprocessingPassPipeline;
   std::string preprocessingTransformSpecFilename;
   std::string preprocessingPDLSpecFilename;
+  TransposeMatmulInput preprocessingTransposeMatmulInput;
```

Comment:
Just going off other examples, it looks like you're intended to pass the pass preprocessing pass pipeline as a string, rather than add custom options for each pre-processing pass, e.g.:
```
--iree-preprocessing-pass-pipeline=builtin.module(func.func(iree-preprocessing-transpose-matmul{input=lhs}))
```

See: https://github.com/iree-org/iree/blob/main/tests/e2e/linalg/BUILD.bazel#L91

Reply:
> I also see there's a iree-preprocessing-transpose-convolution-pipeline, so if we want something more custom I feel adding iree-preprocessing-transpose-matmul-pipeline would be reasonable?

I think so, it's certainly more convenient than the alternative.

Reply:
Putting a note: my understanding is that everything in preprocessing is for fast experiments. It will never be on by default. Otherwise we will move it to GlobalOptimization phase directly. When this pass becomes a need, we will need to revisit how to land it properly in IREE. IMO, the layout changes required by hardware targets can be served by data-tiling.

Reply:
We're on the same page there, as I mentioned on the LLVM patch we're aware there's other options to investigate like data-tiling as you mention, but this gives us a big win for a small amount of effort. Going forward we'll be spending more time looking at the alternatives.